RPIPs/RPIP-60.md

```diff
@@ -1,52 +0,0 @@
----
-rpip: 60
-title: Protocol Upgrade Guardrails
-description: Introduces a delay after protocol upgrades have been confirmed but prior to them coming into effect.
-author: Valdorff (@Valdorff)
-discussions-to: TBD
-status: Draft
-type: Protocol
-category: Core
-created: 2024-03-08
-requires: 
-tags: tokenomics-2024, tokenomics-content
----
-
-## Abstract
-
-Regularly scheduled protocol upgrades now have a delay between when they are passed, and when they are executed. 
-
-The security council may trigger a faster upgrade with a shorter delay, but there are limitations on when they may use this power. This security council power has a maximum number of uses that may be set by the pDAO. The security council cannot use the fast upgrade power if they have no remaining uses, and each use removes one seal. Seals may be restored via pDAO vote. 
-
-Additionally the security council may veto contract upgrades. The veto power should not be used lightly, and misuse may result in replacement of the security council by the pDAO.
-
-## Specification
-
-### Protocol upgrades
-- All protocol upgrades SHALL have a `default_upgrade_delay` delay between when they are passed and when they are executed
-- The security council SHALL have a limited-use power to use a shorter delay of `fast_upgrade_delay`
-  - This power SHALL be usable if `fast_upgrade_seal_count` > 0 
-  - `fast_upgrade_seal_count` SHALL be decremented by one upon using this power
-  - The pDAO MAY include a change to `fast_upgrade_seal_count` in a voted upgrade*
-  - The pDAO MAY change `fast_upgrade_seal_count` via vote**
-  - The security council SHALL use this power if and only if:
-    - A vote specifically requested a fast upgrade
-    - The vote passed with a supermajority of `fast_upgrade_supermajority` or more
-    - The security council deems it acceptable from a security point of view
-- The security council SHALL have the power to veto contract upgrades (which are proposed by the oDAO)
-  - Veto powers SHALL NOT be used lightly and SHALL be reserved for cases of vote manipulation, malicious action (eg, oDAO votes in a proposal against the pDAO governance), or contract upgrades that would result in clear damage to the Rocket Pool project
-  - When this veto power is exercised, the security council MUST publish a Veto Explanation Document that describes why this step was taken; if possible, the report SHOULD also suggest potential similar-but-non-damaging votes that could be considered
-  - If this veto power is abused, the pDAO SHOULD consider replacing the security council
-- The initial settings SHALL be:
-  - `default_upgrade_delay`: 3 weeks
```

Comment:
Does this need to exceed the unstaking delay imposed on RPL staked within nodes? Imo, there should be a way for RPL staking node operators to entirely bail from the protocol before the update takes effect. 

Given having a 6 week delay (2 weeks for visibility, 4 weeks for unstaking delay) seems really long, perhaps the ability to skip the unstaking delay by citing the ID of an upgrade currently queued in the delay system?

Reply:
I'm not sure I'm tracking what "2 weeks for visibility" means. Also note that the vote itself is currently at least 2 weeks long (and only comes after required steps on the forum). I could see the case to change it to 30 days -- just a wee bit of buffer around the minimum. If you're in a weird edge case where you can't make those days work... maybe you start unstaking during the vote -- if it goes your way, you lose 1 month of RPL rewards -- hopefully there aren't many votes where the outcome is the difference between staying and leaving

Reply:
Some more context:

I recently learned that pDAO votes on-chain are a minimum of 3 weeks. `proposal.vote.delay.time` = 1 week between proposal creation and the start of voting, and then 2 weeks of voting. This is for settings and not being 

oDAO proposals right now have 1 week of delay before voting starts and 2 weeks of voting. They can be executed on as soon as quorum is reached (no need to wait to the end of the 2 weeks). In order to be legitimate (and thus not draw a veto from the security council), I'd expect it to be preceded by at least a 1-week snapshot (2-week recommended and I believe always used so far?) and a 1-week sentiment poll.

So it depends a little what we're counting:
- From sentiment poll going up, it's likely 4 weeks plus this setting
- From when the result is known, it is _just_ this setting

So if we want the latter to suffice, we'd need to do 30+ days.
30 days is quite a lot, especially if we get rid of the fast upgrade path (team has concerns about implementation complexity). I wouldn't really fight about it, but it's a real hit imo to be a whole month later to market on every upgrade.

Reply:
Yeah. I don't disagree. It's long. I'm not sure how to reconcile that really. 

Realistically, you _should_ count just from when the result is known. Because exit prior to that is unlikely (people will hope the vote will go their way, and not act.) This is why I suggested skipping the withdrawal period by quoting the ID of a queued upgrade. We could set this to 48 hours then.

Reply:
I don't see the point of that. If you can skip the withdrawal period, it removes much of the benefit of the withdrawal period (smoothing volatility, preventing runs, etc). It might only be a couple times a year, but they're important times.